modules/apps/portal-background-task/portal-background-task-service/src/main/java/com/liferay/portal/background/task/internal/BackgroundTaskMessagingConfigurator.java

```diff
@@ -0,0 +1,141 @@
+
+import com.liferay.portal.background.task.internal.messaging.BackgroundTaskMessageListener;
+import com.liferay.portal.background.task.internal.messaging.BackgroundTaskQueuingMessageListener;
+import com.liferay.portal.background.task.internal.messaging.RemoveOnCompletionBackgroundTaskStatusMessageListener;
+import com.liferay.portal.background.task.service.BackgroundTaskLocalService;
+import com.liferay.portal.kernel.backgroundtask.BackgroundTaskExecutorRegistry;
+import com.liferay.portal.kernel.backgroundtask.BackgroundTaskStatusRegistry;
+import com.liferay.portal.kernel.backgroundtask.BackgroundTaskThreadLocalManager;
+import com.liferay.portal.kernel.lock.LockManager;
+import com.liferay.portal.kernel.messaging.Destination;
+import com.liferay.portal.kernel.messaging.DestinationConfiguration;
+import com.liferay.portal.kernel.messaging.DestinationFactory;
+import com.liferay.portal.kernel.messaging.DestinationNames;
+import com.liferay.portal.kernel.messaging.MessageBus;
+import com.liferay.portal.kernel.util.HashMapDictionaryBuilder;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Vendel Toreki
+ */
+@Component(
+	immediate = true, service = BackgroundTaskMessagingConfigurator.class
+)
+public class BackgroundTaskMessagingConfigurator {
+
+	@Activate
+	protected void activate(BundleContext bundleContext) {
+		_bundleContext = bundleContext;
+
+		Destination backgroundTaskDestination = _registerDestination(
+			bundleContext, DestinationConfiguration.DESTINATION_TYPE_PARALLEL,
+			DestinationNames.BACKGROUND_TASK, 5, 10);
+
+		backgroundTaskDestination.register(
+			new BackgroundTaskMessageListener(
+				_backgroundTaskExecutorRegistry, _backgroundTaskLocalService,
+				_backgroundTaskStatusRegistry,
+				_backgroundTaskThreadLocalManager, _lockManager, _messageBus));
+
+		Destination backgroundTaskStatusDestination = _registerDestination(
+			bundleContext, DestinationConfiguration.DESTINATION_TYPE_SERIAL,
+			DestinationNames.BACKGROUND_TASK_STATUS, 1, 1);
+
+		backgroundTaskStatusDestination.register(
+			new BackgroundTaskQueuingMessageListener(
+				_backgroundTaskLocalService, _lockManager));
+		backgroundTaskStatusDestination.register(
+			new RemoveOnCompletionBackgroundTaskStatusMessageListener(
+				_backgroundTaskLocalService));
+	}
+
+	@Deactivate
+	protected void deactivate() {
+		for (ServiceRegistration<Destination> serviceRegistration :
+				_serviceRegistrations) {
+
+			Destination destination = _bundleContext.getService(
+				serviceRegistration.getReference());
+
+			serviceRegistration.unregister();
+
+			destination.destroy();
```

Comment:
@rafaprax Destination is opened/closed by DefaultMessageBus upon register/unregister the service, you should not do it again here.